applications/plugins/SofaSphFluid/src/SofaSphFluid/SPHFluidForceField.inl

```diff
@@ -164,13 +170,30 @@ void SPHFluidForceField<DataTypes>::computeNeighbors(const core::MechanicalParam
                 }
             }
         }
+#else
+        std::for_each(std::execution::par, x.begin(), x.end(), [&](const auto& ri)
+        {
+            int i = &ri - &x[0]; // only possible with vector, etc.
```

Comment:
use `std::distance` here?

Reply:
nope, you need iterators for std::distance(), and for_each gives you the value directly ("ri" here is not an iterator)